go/vt/schemadiff/table.go

```diff
@@ -1731,6 +1731,40 @@ func evaluateColumnReordering(t1SharedColumns, t2SharedColumns []*sqlparser.Colu
 	return minimalColumnReordering
 }
 
+// This function looks for a non-deterministic function call in the given expression.
+// If recurses into all function arguments.
+// The known non-deterministic function we handle are:
+// - UUID()
+// - RAND()
+// - SYSDATE()
+func findNoNondeterministicFunction(expr sqlparser.Expr) (foundFunction string) {
+	_ = sqlparser.Walk(func(node sqlparser.SQLNode) (kontinue bool, err error) {
+		switch node := node.(type) {
+		case *sqlparser.CurTimeFuncExpr:
+			switch node.Name.Lowered() {
+			case "sysdate":
+				foundFunction = node.Name.String()
+				return false, nil
+			}
+		case *sqlparser.FuncExpr:
+			switch node.Name.Lowered() {
+			case "uuid", "rand":
+				foundFunction = node.Name.String()
+				return false, nil
+			default:
+				// recurse into function argument expressions
```

Comment:
Is this recursion needed? Walk already visits everything doesn't it?

Reply:
Confirmed, the recursion is not needed. An earlier iteration led me to believe it was. The existing unit tests already cover two inner function scenarios so this is testes.